c10/core/TensorImpl.cpp

```diff
@@ -885,6 +885,33 @@ void TensorImpl::ShareExternalPointer(
   }
 }
 
+bool _compute_contiguous(const ExtraMeta& extra_meta, std::vector<int> order) {
+  if (order.size() != extra_meta.sizes_.size())
+    return false;
+  bool is_contiguous = true;
+  if (extra_meta.numel_ == 0)
+    return is_contiguous;
+  SymInt z = 1;
+  for (auto d : order) {
+    const auto size_d = extra_meta.sizes_.at(d);
+    if (size_d != 1) {
+      if (extra_meta.strides_.at(d) == z) {
+        z *= size_d;
+      } else {
+        is_contiguous = false;
+        break;
+      }
+    }
+  }
+  return is_contiguous;
+}
```

Comment:
note to reviewer: need to make sure this is synchronized with the other compute contiguous